### ISSUE TITLE
Grid2OpException DivergingPowerFlow

### ISSUE THREAD

**Issue by ML-IEE** [rte-france/Grid2Op#362]
## Environment
 - Grid2op version: 1.7.1
 - System: CentOS Linux 7
 - Ligthsim2grid: 0.7.0

## Bug description
An Agent runs through the Grid2Op Environment and only changes the topology with 'set_bus_vect'. 
After some runs the agents encounters a DivergingPowerFlow error with Grid2OpException DivergingPowerFlow DivergingPowerFlow('Divergence of DC powerflow (non connected grid) at the initialization of AC powerflow. Detailed error: ErrorType.SolverFactor'. 

In other cases the error was: 
Grid2OpException DivergingPowerFlow DivergingPowerFlow('Divergence of AC powerflow. Detailed error: ErrorType.TooManyIterations')

Is this Exeption within the scope/learning process of the agent or is this some internal problem. Note, that this also was reproducible with the pandapower backend. 

## How to reproduce
Added a pickle document containing exemplary actions that should reproduce the problem. 

### Code snippet
```python
from lightsim2grid import LightSimBackend
import grid2op
import pickle 
import os 
import zipfile


with zipfile.ZipFile("actions.zip", 'r') as zip_ref:
    zip_ref.extractall(os.getcwd())

with open('actions.pkl', 'rb') as f:
    steps = pickle.load( f)


    
backend = LightSimBackend()
env = grid2op.make("l2rpn_neurips_2020_track1_small",backend=backend)
env.set_id(1)
env.reset()
env.chronics_handler.get_name()
obs = env.get_obs()



i = 0 
done = False
while not done:
    act = env.action_space.from_vect(steps[i])
    obs,rew,done,info = env.step(act)
    i += 1

    
print(info)

```

## Current output
```
{'disc_lines': array([-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,  0,
       -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
        0,  0, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
       -1, -1, -1, -1, -1, -1, -1, -1], dtype=int32), 'is_illegal': False, 'is_ambiguous': False, 'is_dispatching_illegal': False, 'is_illegal_reco': False, 'reason_alarm_illegal': None, 'opponent_attack_line': None, 'opponent_attack_sub': None, 'opponent_attack_duration': 0, 'exception': [Grid2OpException DivergingPowerFlow DivergingPowerFlow('Divergence of DC powerflow (non connected grid) at the initialization of AC powerflow. Detailed error: ErrorType.SolverFactor')], 'rewards': {}}
```

[actions.zip](https://github.com/rte-france/Grid2Op/files/9148106/actions.zip)

**Comment by BDonnot** [comment]
Hello,

Sorry I thought I answered this issue but apparently I did not.

Diverging powerflow is perfectly normal. It can happen when the grid is not connected (for example a load or a generator is not connected to the main grid), or because there is no feasible solution for the Kirchhoff Current Laws.

In both cases this is a game over in grid2op. In real grid this would probably mean that the grid collapsed at some point.

Hope I answered your question

Best

Benjamin
